src/multipole_header.F90

```diff
@@ -42,87 +46,142 @@ module multipole_header
+
+    character(len=10) :: version
+    integer :: i, n_poles, n_residue_types, n_windows
+    integer(HSIZE_T) :: dims_1d(1), dims_2d(2), dims_3d(3)
+    integer(HID_T) :: file_id
+    integer(HID_T) :: group_id
+    integer(HID_T) :: dset
+    type(DictIntInt) :: l_val_dict
+
+    ! Open file for reading and move into the /isotope group
+    file_id = file_open(filename, 'r', parallel=.true.)
+    group_id = open_group(file_id, "/nuclide")
+
+    ! Check the file version number.
+    call read_dataset(version, file_id, "version")
+    if (version /= VERSION_MULTIPOLE) call fatal_error("The current multipole&
+         & format version is " // trim(VERSION_MULTIPOLE) // " but the file "&
+         // trim(filename) // " uses version " // trim(version) // ".")
+
+    ! Read scalar values.
+    call read_dataset(this % formalism, group_id, "formalism")
+    call read_dataset(this % spacing, group_id, "spacing")
+    call read_dataset(this % sqrtAWR, group_id, "sqrtAWR")
+    call read_dataset(this % start_E, group_id, "start_E")
+    call read_dataset(this % end_E, group_id, "end_E")
+
+    ! Read the "data" array.  Use its shape to figure out the number of poles
+    ! and residue types in this data.
+    dset = open_dataset(group_id, "data")
+    call get_shape(dset, dims_2d)
+    n_residue_types = int(dims_2d(1), 4) - 1
+    n_poles = int(dims_2d(2), 4)
+    allocate(this % data(n_residue_types+1, n_poles))
+    call read_dataset(this % data, dset)
+    call close_dataset(dset)
+
+    ! Check to see if this data includes fission residues.
+    if (this % formalism == FORM_RM) then
+      this % fissionable = (n_residue_types == 3)
+    else
+      ! Assume FORM_MLBW.
+      this % fissionable = (n_residue_types == 4)
+    end if
+
+    ! Read the "l_value" array.
+    allocate(this % l_value(n_poles))
+    call read_dataset(this % l_value, group_id, "l_value")
+
+    ! Figure out the number of unique l values in the l_value array.
+    do i = 1, n_poles
+      if (.not. l_val_dict % has(this % l_value(i))) then
+        call l_val_dict % set(this % l_value(i), 0)
       end if
-    end subroutine
+    end do
+    this % num_l = l_val_dict % size()
+    call l_val_dict % clear()
+
+    ! Read the "pseudo_K0RS" array.
+    allocate(this % pseudo_k0RS(this % num_l))
+    call read_dataset(this % pseudo_k0RS, group_id, "pseudo_K0RS")
+
+    ! Read the "w_start" array and use its shape to figure out the number of
+    ! windows.
+    dset = open_dataset(group_id, "w_start")
+    call get_shape(dset, dims_1d)
+    n_windows = int(dims_1d(1), 4)
+    allocate(this % w_start(n_windows))
+    call read_dataset(this % w_start, dset)
+    call close_dataset(dset)
+
+    ! Read the "w_end" and "broaden_poly" arrays.
+    allocate(this % w_end(n_windows))
+    call read_dataset(this % w_end, group_id, "w_end")
+    allocate(this % broaden_poly(n_windows))
+    call read_dataset(this % broaden_poly, group_id, "broaden_poly")
```

Comment:
I'm skeptical this will still work if `broaden_poly` was written as an array of `np.bool_`. From what I understand, in this case, h5py will write it out using an enum. Have you tried writing out multipole data using the new `export_to_hdf5` method and then using it in a simulation?

Reply:
Thinking about this more -- probably the simplest solution is to use an array of bools in the `WindowedMultipole` Python object but to write out/read in an array of ints. If you do stick with writing bools (enums), the io_formats doc should be updated to reflect that.

Reply:
FYI, the Fortran code happily accepted the enums and worked fine.  The enum was 0 for false and 1 for true so reading them as integers had the same effect.  Nevertheless, we might not be able to depend on that behavior so I've converted `broaden_poly` to an array of 8-bit integers on write.